src/main/java/org/socialculture/platform/ticket/service/TicketServiceImpl.java

```diff
@@ -83,21 +87,52 @@ public TicketResponseDto getTicketByEmailAndTicketId(String email, Long ticketId
     @Override
     @Transactional
     public TicketResponseDto registerTicket(String email, TicketRequestDto ticketRequest) {
-        // 회원 검색
+        log.info("티켓 등록 트랜잭션 시작, 사용자 이메일: " + email);
+
         MemberEntity memberEntity = findMemberByEmail(email);
 
-        // 공연 검색
-        PerformanceEntity performanceEntity = findPerformanceById(ticketRequest.performanceId());
+        log.info("공연에 대한 락을 획득 시도 중, 공연 ID: " + email);
+
+        PerformanceEntity performanceEntity;
+        try {
+            performanceEntity = performanceRepository.findByIdWithLock(ticketRequest.performanceId())
+                    .orElseThrow(() -> new GeneralException(ErrorStatus.PERFORMANCE_NOT_FOUND));
+        } catch (PessimisticLockException e) {
+            log.error("공연 락 획득 실패, 공연 ID: " + email);
+            throw e;
+        }
+
+       log.info("공연 락 획득 완료, 공연 ID: " + email);
+
+        if (performanceEntity.getRemainingTickets() < ticketRequest.quantity()) {
+            log.warn("남은 티켓 수 부족, 공연 ID: " + ticketRequest.performanceId());
+            throw new GeneralException(ErrorStatus._NOT_ENOUGH_TICKETS);
+        }
+        //비관적 락이 걸린 후에 대기 시간 추가 (시각적으로 확인하기 위함)
+//        try {
+//            Thread.sleep(10000);  // 5초 동안 대기
+//        } catch (InterruptedException e) {
+//            Thread.currentThread().interrupt();
+//            throw new RuntimeException(e);
+//        }
 
-        // 가격 계산 : 가격, 예매인원, 쿠폰 아이디 전달 -> 쿠폰이 있다면 할인율 까지 계산
         int finalPrice = calculateFinalPrice(performanceEntity.getPrice(), ticketRequest.quantity(), ticketRequest.couponId());
 
-        // 티켓 생성 및 저장
         TicketEntity ticketEntity = createAndSaveTicket(memberEntity, performanceEntity, ticketRequest.quantity(), finalPrice);
 
```

Comment:
```suggestion
private updateRemainingTickets(PerformanceEntity performanceEntity, int ticketQuantity) {
    int remainTicket = performanceEntity.getRemainingTickets() - ticketEntity.getQuantity();
    log.info("티켓 등록 후 남은 티켓 수량: " + remainTicket);

    performanceEntity.updateTicket(remainTicket);
    log.info("공연 테이블의 남은 티켓 수 업데이트 완료: " + performanceEntity.getRemainingTickets());
}
```
이 부분도 공연의 티켓을 수정한다는 개념으로 따로 분리가 되어있으면 좋을 것 같아요!